src/lib.rs

```diff
@@ -182,6 +182,31 @@ impl Error for ParseBigIntError {
     }
 }
 
+#[cfg(has_try_from)]
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq, Hash)]
```

Comment:
I was just thinking that we might want to consider storing the original value in the error for `TryFrom<BigInt>`, but `Copy` would prevent that. We don't have to resolve that consideration right now, but let's remove `Copy` so we could do it later.

Also, `Default` and `Hash` seem questionable -- the standard `TryFromIntError` doesn't have those.

Reply:
Storing the original value makes sense. However, currently the error's thrown at every point the conversion fails, so the original value can be one of multiple types: `BigInt`, `BigUint`, and maybe some primitives if we're storing those too.

Some possible solutions:

* Make the error type generic over original type: `TryFromBigIntError<T>`. I think this requires change now to be non-breaking.
* One error type for each original type; Actually a more verbose version of generic solution. Also requires change now.
* Store an enum of possible original types. Could be future-proofed by making the enum non-exhaustive. Has some runtime cost. I think keeping the current design ties us to this one, but I could be wrong.

So I believe this decision needs to be made before release; IMO going generic makes the most sense, but again I could be overlooking something.

Reply:
> so the original value can be one of multiple types: `BigInt`, `BigUint`, and maybe some primitives if we're storing those too.

Also `&BigInt` and `&BigUint`, but I don't think there's much value in storing those, nor primitives. A lot of `dyn Error` APIs want `'static`, which makes references inconvenient.

So that would make 3 error categories: `TryFrom<BigInt>`, `TryFrom<BigUint>`, and the rest unsaved. I'm also leaning toward the generic solution, and I guess the unsaved can use `<()>`. Then the saved cases each need an extraction method, like `fn into_inner(self) -> BigInt`. I think those two can overlap the same method name.

> Store an enum of possible original types. Could be future-proofed by making the enum non-exhaustive.

I would want to keep an enum private, like parsing's `BigIntErrorKind` now, although it's also awkward to take different types out -- I guess it would need multiple methods returning `Option`.

Reply:
I agree that the enum API would be rather awkward. I'll work on the generic design.

Reply:
And it's done(406f192 and 0da7ae7). Specifically:

* The error is now generic over pre-conversion type
* Error sites modified to return correct error value
* `Error` impl now requires `T: Debug`
* Some docs on the error type added

The `Error` impl issue shouldn't matter much, since all our current `T`s implement `Debug`. It would be nicer if `T: !Debug` cases could be handled without making the whole Debug impl less useful, but I couldn't figure out how.